File: python/124나라의숫자.py
def solution(n):
    answer = ''
    append_num = ['4', '1', '2']
    while n != 0:
        answer = append_num[n%3] + answer
        if n % 3 == 0:
            n = int(n/3) - 1
        else:
            n = int(n/3)
    return answer

print(solution(11))
print(solution(12))
print(solution(13))
print(solution(14))

File: JavaScript/124나라의숫자.js
function solution(n) {
    let answer = '';
    let remainder;
    while (n >= 1) {
        remainder = n % 3;
        n = Math.floor(n / 3);
        if (remainder === 1) {
            answer = '1' + answer;
        } else if (remainder === 2) {
            answer = '2' + answer;
        } else {
            answer = '4' + answer;
            n -= 1;
        }
    }
    return answer;
}

console.log(solution(500000000000))